src/ip.cpp

```diff
@@ -4,6 +4,21 @@
 
 namespace oxen::quic
 {
+    ipv4::ipv4(const std::string& str)
+    {
+        struct sockaddr_in sin
+        {};
+        sin.sin_family = AF_INET;
+        auto rv = inet_pton(AF_INET, str.c_str(), &sin.sin_addr);
+
+        if (rv == 0)  // inet_pton returns this on invalid input
+            throw std::invalid_argument{"Cannot construct address: invalid IP"};
```

Comment:
This message seems off since this isn't constructing an address: perhaps just `"Unable to parse IP address"`